pkg/controller/servicebindingrequest/reconciler.go

```diff
@@ -194,6 +200,31 @@ func (r *Reconciler) Reconcile(request reconcile.Request) (reconcile.Result, err
 		return sb.Unbind()
 	}
 
+	var emptyApplication v1alpha1.ApplicationSelector
+	if sbr.Spec.ApplicationSelector == emptyApplication ||
+		(sbr.Spec.ApplicationSelector.ResourceRef == "" &&
+			sbr.Spec.ApplicationSelector.LabelSelector.MatchLabels == nil) {
+		conditionsv1.SetStatusCondition(&sbr.Status.Conditions,
+			conditionsv1.Condition{
+				Type:    InjectionReady,
+				Status:  corev1.ConditionFalse,
+				Reason:  EmptyApplicationSelectorReason,
+				Message: "The spec.applicationSelector field is empty.",
+			},
+		)
+		conditionsv1.SetStatusCondition(&sbr.Status.Conditions,
+			conditionsv1.Condition{
+				Type:   CollectionReady,
+				Status: corev1.ConditionTrue,
+			},
+		)
+		_, updateErr := updateServiceBindingRequestStatus(r.dynClient, sbr)
+		if updateErr == nil {
+			return Done()
+		}
+		return Done()
```

Comment:
return Done() is there in if condition also and after that too, then what would be the purpose of if updateErr==nil

Reply:
Resolved 
02eb80f